packages/astro/src/runtime/server/jsx.ts

```diff
@@ -64,7 +66,10 @@ Did you forget to import the component or is it possible there is a typo?`);
 						props[key] = value;
 					}
 				}
-				const str = await renderToString(result, vnode.type as any, props, slots);
```

Comment:
This was calling the component factory as is. Instead, we want it to go through `renderAstroComponent` so the server island special handling can take effect